DFE-Digital/ecf-engage-and-learn
# Add sign-in by magic links for admin users

## Ticket and context

Ticket: https://dfedigital.atlassian.net/browse/CPDEL-453

We want to hide admin users behind proper authentication. 
Since we use staging environment as our source of truth for content, it is really convenient to keep a dummy admin without that requirement in there, but I added a second admin for whom we can test that email sending works.

## Code review

### Is there anything weird that code reviewer should know?
Our `passwordless_authenticatable` was serving largely as email validator. I decided to move it to the controller.

I copied most of this stuff from R&P, but there are some differences. For one, I think it's much more convenient to specify the test users in the strategy, rather than controller. The other is the obvious one, ects and mentors are still logged in without authentication.

One more change I copied from R&P is to have signed out page in the sessions controller, with all other user sign-in actions.

There are some changes in cypress - cause the login flow for admins is fundamentally different to the one for other users. Maybe I could check if traits include admin to avoid adding more steps. I'll let you judge whether it's a better idea.

### Review Checks

## Product review

### How can someone see it it review app?
1. Click the link to review app (posted by a `github-actions` bot below)
2. Try logging in as `ecf-engage-and-learn@mailinator.com`
3. Go to https://www.mailinator.com/v4/public/inboxes.jsp?to=ecf-engage-and-learn
4. There should be a sign-in email there, and when you click on sign-in link you should be able to log in.
